presto/dataset.py

```diff
@@ -227,7 +227,58 @@ def __getitem__(self, idx):
             real_mask_per_variable,
         )
 
+    
+class WorldCerealMasked10DDataset(WorldCerealBase):
```

Comment:
nit: this can extend `WorldCerealMaskedDataset` instead so you don't need to reimplement methods:

```python
WorldCerealMasked10DDataset(WorldCerealMaskedDataset):
    NUM_TIMESTEPS = 36

    @classmethod
    def get_month_array(cls, row: pd.Series) -> np.ndarray:
        start_date, end_date = datetime.strptime(row.start_date, "%Y-%m-%d"), datetime.strptime(
            row.end_date, "%Y-%m-%d"
        )

        # Calculate the step size for 10-day intervals and create a list of dates
        step = int((end_date - start_date).days / (cls.NUM_TIMESTEPS - 1))
        date_vector = [start_date + timedelta(days=i * step) for i in range(cls.NUM_TIMESTEPS)]

        # Ensure last date is not beyond the end date
        if date_vector[-1] > end_date:
            date_vector[-1] = end_date

        return np.array([d.month - 1 for d in date_vector])
```